var/spack/repos/builtin/packages/py-onnxruntime/package.py

```diff
@@ -50,6 +77,7 @@ class PyOnnxruntime(CMakePackage, PythonExtension):
     depends_on("py-coloredlogs", when="@1.17:", type=("build", "run"))
     depends_on("py-flatbuffers", type=("build", "run"))
     depends_on("py-numpy@1.16.6:", type=("build", "run"))
+    depends_on("py-numpy@1.21.6:1", when="@1.18", type=("build", "run"))
```

Comment:
I'd split this, since the upper limit likely applies before as well (not sure if 1.19 still has the restriction).
```suggestion
    depends_on("py-numpy@1.21.6:", when="@1.18", type=("build", "run"))
    depends_on("py-numpy@:1", when="@:1.18", type=("build", "run"))
```

Reply:
It looks like 1.19 has support for numpy 2 so the upper limit won't apply for it. It's lower limit will remain 1.21.6. 
https://github.com/microsoft/onnxruntime/blob/v1.19.0/requirements.txt